ouroboros-network/test/Test/ChainFragment.hs

```diff
@@ -372,6 +373,9 @@ prop_toChain_fromChain :: TestBlockChain -> Property
 prop_toChain_fromChain (TestBlockChain c) =
     CF.toChain (CF.fromChain c) === c
 
+prop_fixupBlock' :: TestBlockChainFragment -> Bool
+prop_fixupBlock' (TestBlockChainFragment chain) =
+  fixupChainFragment fixupBlock' (CF.toNewestFirst chain) == chain
```

Comment:
Doesn't this demonstrate that `fixupBlock'` already does the right thing for chains that start from genesis? So then we don't need two variants?